### PR TITLE
initial version

### PR DESCRIPTION
**Pull request by mwennrich**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+FROM golang:1.17-alpine as builder
```
Comment (majst01) [inline Dockerfile:2]: I would go for 1.18 right from the beginning

**Comment by majst01** [comment]
Top !

File: .github/workflows/latest.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+name: Build latest image
```
Comment (Gerrit91) [inline .github/workflows/latest.yaml:1]: If you are interested, we can also try a new version that I came up with. Should work and only requires a single actions file:

```yaml
---
name: Docker Build Action
on:
  pull_request:
    branches:
      - main
  release:
    types:
      - published
  push:
    branches:
      - main

env:
  REGISTRY: ghcr.io
  IMAGE_NAME: ${{ github.repository }}

jobs:
  build:
    name: Build
    runs-on: ubuntu-latest
    steps:
    - name: Set up Go 1.18
      uses: actions/setup-go@v2
      with:
        go-version: 1.18

    - name: Check out code into the Go module directory
      uses: actions/checkout@v2

    - name: Lint
      uses: golangci/golangci-lint-action@v2
      with:
        args: -p bugs -p unused --timeout=3m

    - name: Log in to the Container registry
      uses: docker/login-action@v1
      with:
        registry: ${{ secrets.DOCKER_REGISTRY }}
        username: ${{ secrets.DOCKER_REGISTRY_USER }}
        password: ${{ secrets.DOCKER_REGISTRY_TOKEN }}

    - name: Make tag
      run: |
        [ "${GITHUB_EVENT_NAME}" == 'pull_request' ] && echo "tag=${GITHUB_HEAD_REF##*/}" >> $GITHUB_ENV || true
        [ "${GITHUB_EVENT_NAME}" == 'release' ] && echo "tag=${GITHUB_REF##*/}" >> $GITHUB_ENV || true
        [ "${GITHUB_EVENT_NAME}" == 'push' ] && echo "tag=latest" >> $GITHUB_ENV || true

    - name: Build and push reporter image
      uses: docker/build-push-action@v2
      with:
        context: .
        push: true
        tags: ${{ secrets.DOCKER_REGISTRY }}/${{ env.IMAGE_NAME }}:${{ env.tag }}
```

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,62 @@
+module github.com/metal-stack/metal-metrics-exporter
+
+go 1.18
+
+require (
+	github.com/metal-stack/metal-go v0.16.3
```
Comment (Gerrit91) [inline go.mod:6]: Should be 0.16.5 by now.

**Pull request merged**